File: src/pooh3/mobi/app/a01_clock/model/Clock.java
package pooh3.mobi.app.a01_clock.model;


import static pooh3.mobi.Preconditions.checkNotNull;

public class Clock {
    public final ShortHand shortHand;
    public final LongHand longHand;

    private Clock(ShortHand shortHand, LongHand longHand) {
        this.shortHand = checkNotNull(shortHand);
        this.longHand = checkNotNull(longHand);
    }

    public static Clock build(ShortHand shortHand, LongHand longHand) {
        return new Clock(shortHand, longHand);
    }

    @Override
    public String toString() {
        return shortHand.hour + ":" + longHand.minute;
    }
}

File: src/pooh3/mobi/app/a01_clock/model/Clock.kt
package pooh3.mobi.app.a01_clock.model

import pooh3.mobi.checkNotNull


class Clock private constructor(shortHand: ShortHand, longHand: LongHand) {
    val shortHand: ShortHand = shortHand.checkNotNull
    val longHand: LongHand = longHand.checkNotNull

    companion object {

        fun build(shortHand: ShortHand, longHand: LongHand): Clock {
            return Clock(shortHand, longHand)
        }
    }
}
